python/__init__.py

```diff
@@ -511,35 +536,58 @@ def set_uart_callback(self, uart, install):
   # Timeout is in ms. If set to 0, the timeout is infinite.
   CAN_SEND_TIMEOUT_MS = 10
 
-  def can_send_many(self, arr, timeout=CAN_SEND_TIMEOUT_MS):
-    snds = []
-    transmit = 1
-    extended = 4
-    for addr, _, dat, bus in arr:
+  def len_to_dlc(self, length):
+    if length <=8:
+      return length
+    if length <=24:
+      return 8 + ((length - 8) // 4) + (1 if length % 4 else 0)
+    else:
+      return 11 + (length // 16) + (1 if length % 16 else 0)
+
+  def pack_can_buffer(self, arr):
```

Comment:
as mentioned in other comments, let's make a pack and unpack function, add a unit test for these and have this function just do a `list(map(pack, msgs))` or something like that